test/src/unit-capi-config.cc

```diff
@@ -724,6 +724,7 @@ TEST_CASE("C API: Test config iter", "[capi][config]") {
   all_param_values["vfs.s3.connect_timeout_ms"] = "10800";
   all_param_values["vfs.s3.connect_max_tries"] = "5";
   all_param_values["vfs.s3.connect_scale_factor"] = "25";
+  all_param_values["vfs.s3.custom_headers."] = "";
```

Comment:
We should remove this config entry as there's no need for it any more with the iterator approach to key/value discovery.